pkg/imagestream/imagestream.go

```diff
@@ -0,0 +1,186 @@
+package imagestream
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	imagev1 "github.com/openshift/api/image/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Builder provides a struct for imageStream object from the cluster and a imageStream definition.
+type Builder struct {
+	// imageStream definition, used to create the imageStream object.
+	Definition *imagev1.ImageStream
+	// Created imageStream object.
+	Object *imagev1.ImageStream
+	// api client to interact with the cluster.
+	apiClient goclient.Client
+	// Used in functions that define or mutate clusterOperator definition. errorMsg is processed before the
+	// ClusterOperator object is created.
+	errorMsg string
+}
+
+// Pull retrieves an existing imageStream object from the cluster.
+func Pull(apiClient *clients.Settings, name, nsname string) (*Builder, error) {
+	glog.V(100).Infof(
+		"Pulling imageStream object name %s from namespace %s", name, nsname)
+
+	if apiClient == nil {
+		glog.V(100).Infof("The apiClient is empty")
+
+		return nil, fmt.Errorf("imageStream 'apiClient' cannot be empty")
+	}
+
```

Comment:
could you please add an AttachScheme here like in [mcp/mcp.go](https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/mco/mcp.go#L86-L91)? Unfortunately this won't get caught by our tests but it'll cause errors when you go to use the function if you don't include it

Reply:
done